### PR TITLE
feat(rollup, coordinator, bridge-history-api): Allow Environment Variables to override config.json

### PR DESCRIPTION
**Pull request by dghelm**
### Purpose or design rationale of this PR

The goal is to allow services to be configured using environmental variables (in addition to the config.json file). This facilitates contexts like Kubernetes, where you want Secrets managed independently from the non-sensitive config information (say, loaded from AWS Secrets).

Here, variables are prefixed with `SCROLL_[SERVICE]_`. 

Sensitive values include:
- `SCROLL_ROLLUP_DB_CONFIG_DSN`
- `SCROLL_ROLLUP_L1_CONFIG_RELAYER_CONFIG_GAS_ORACLE_SENDER_PRIVATE_KEY`
- `SCROLL_ROLLUP_L2_CONFIG_RELAYER_CONFIG_GAS_ORACLE_SENDER_PRIVATE_KEY`
- `SCROLL_ROLLUP_L2_CONFIG_RELAYER_CONFIG_COMMIT_SENDER_PRIVATE_KEY`
- `SCROLL_ROLLUP_L2_CONFIG_RELAYER_CONFIG_FINALIZE_SENDER_PRIVATE_KEY`
- `SCROLL_BRIDGE_HISTORY_DB_DSN`
- `SCROLL_BRIDGE_HISTORY_REDIS_PASSWORD`
- `SCROLL_COORDINATOR_AUTH_SECRET`
- `SCROLL_COORDINATOR_DB_DSN`

I have not run this code in the Scroll SDK stack, and only a minimal test has been run to see if it works. This at least shows the string is being used as though it came from the JSON file.

```
❯ export SDKROLLUP_DB_CONFIG_DSN=postgres://postgres:azerty12345@postgresql-rollup:5432/scroll?sslmode=disable
❯ ./build/bin/gas_oracle --config ./conf/config.json
ERROR[08-20|13:44:18.305] gorm                                     err message="failed to initialize database, got error failed to connect to `host=postgresql-rollup user=postgres database=scroll`: hostname resolving error (lookup postgresql-rollup on 10.255.255.254:53: read udp 10.255.255.254:54234->10.255.255.254:53: i/o timeout)"
CRIT [08-20|13:44:18.305] failed to init db connection             err="failed to connect to `host=postgresql-rollup user=postgres database=scroll`: hostname resolving error (lookup postgresql-rollup on 10.255.255.254:53: read udp 10.255.255.254:54234->10.255.255.254:53: i/o timeout)"
```

**Edited to include additional services.**
Also see:
- https://github.com/scroll-tech/chain-monitor/pull/125
- https://github.com/scroll-tech/scroll-admin-system/pull/32

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [x] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [ ] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

**Comment by georgehao** [comment]
BTW, what's the rationale of scroll-sdk needing ENV ? since there are some many options for rollup_realyer and gas_oracle

**Comment by dghelm** [comment]
Adjusted prefix to SCROLL_ROLLUP to not be SDK specific.

**Review by colinlyguo — DISMISSED** [review]

File: common/utils/utils.go [hunk 1]
```diff
@@ -78,3 +81,88 @@ func ReadGenesis(genesisPath string) (*core.Genesis, error) {
...
+					return err
+				}
+			}
+		case reflect.Struct:
+			err := OverrideConfigWithEnv(fieldValue.Addr().Interface(), envKey)
+			if err != nil {
+				return err
+			}
+		default:
+			if envValue, exists := os.LookupEnv(envKey); exists {
```
Comment (Thegaram) [inline common/utils/utils.go:124]: This log will make debugging much easier.

```suggestion
			if envValue, exists := os.LookupEnv(envKey); exists {
				log.Info("Overriding config with env var", "key", envKey)
```

**Review by Thegaram — DISMISSED** [review]

File: common/utils/utils.go (hunk 1 above)
Comment (colinlyguo) [reply common/utils/utils.go:124]: A side effect: it will also print the private key, could check the use cases (if it's acceptable).

Comment (Thegaram) [reply common/utils/utils.go:124]: That's why I suggest to only print the key and not the value.

Comment (colinlyguo) [reply common/utils/utils.go:124]: oh I see.

**Review by colinlyguo — APPROVED** [review]

**Review by yiweichi — APPROVED** [review]
LGTM

**Pull request merged**